### PR TITLE
Onboard repo to Secure Development Tools Azure DevOps

### PR DESCRIPTION
**Pull request by TinaMor**
## PR Description
This PR onboards this repo to Secure Development Tools used for running of security and compliance static analysis tools. It adds a YAML file that will be used to create an Azure DevOps pipeline. This pipeline will be triggered by PRs created against the main branch.

### Reference
1. CodeQL (previously Semmle)
    - [CodeQL Portal](https://semmleportal.azurewebsites.net/)
    - [CodeQL for YAML-based Azure Pipelines](https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/codeql/codeql-for-yaml-based-azure-pipelines)
    - [CodeQL Build Task](https://eng.ms/docs/microsoft-security/security/azure-security/cloudai-security-fundamentals-engineering/security-integration/guardian-wiki/sdl-azdo-extension/codeql-build-task)
    - [Recommended hardware resources for running CodeQL](https://docs.github.com/en/enterprise-cloud@latest/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/recommended-hardware-resources-for-running-codeql)

2. CodeInspector
    - [CodeInspector Build Task](https://eng.ms/docs/microsoft-security/security/azure-security/cloudai-security-fundamentals-engineering/security-integration/guardian-wiki/sdl-azdo-extension/codeinspector-build-task)

3. Publish Security Analysis Logs
    - [Publish Security Analysis Logs Build Task](https://eng.ms/docs/microsoft-security/security/azure-security/cloudai-security-fundamentals-engineering/security-integration/guardian-wiki/sdl-azdo-extension/publish-security-analysis-logs)

### Sample Run

image

### CODE REVIEW

**Review by profnandaa — COMMENTED** [review]
wondering if this should be added as part of the [`/azure-pipelines.yml`](https://github.com/microsoft/windows-container-tools/blob/main/azure-pipelines.yml) instead?

**Comment by profnandaa** [comment]
> wondering if this should be added as part of the [`/azure-pipelines.yml`](https://github.com/microsoft/windows-container-tools/blob/main/azure-pipelines.yml) instead?

Approving; as you've explained about decoupling the pipeline since it takes a little longer that the current one.

**Review by profnandaa — DISMISSED** [review]

File: sdl-compliance-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      env:
+        SYSTEM_ACCESSTOKEN: $(System.AccessToken)
+      inputs:
+        sourceCodeDirectory: '$(Build.SourcesDirectory)\LogMonitor'
+        language: 'cpp'
+        buildCommands: '$(msbuildPath) $(BUILD.SourcesDirectory)\$(solution) /t:clean #
+                        $(msbuildPath) $(BUILD.SourcesDirectory)\$(solution) /p:platform=$(BuildPlatform) /p:configuration=$(BuildConfiguration)'
+        querySuite: 'Recommended'
+        timeout: '1800'
+        ram: '16384'
```
Comment (iankingori) [inline sdl-compliance-pipeline.yml:37]: is this 16GB? Why so much ram?

Comment (bobsira) [reply sdl-compliance-pipeline.yml:37]: Part of those performance initiatives around LogMonitor could include investigating the minimum system resources to build Log Monitor.

**Review by bobsira — APPROVED** [review]
LGTM!

File: sdl-compliance-pipeline.yml (hunk 1 above)
Comment (bobsira) [reply sdl-compliance-pipeline.yml:37]: From the sample CodeQL [codes](https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/codeql/codeql-for-yaml-based-azure-pipelines#:~:text=ram%3A-,16384,-Required) 16384 is the value provided maybe if it worth exploring if we want to half this to save on resources but I do not see any warnings against having this.

Comment (TinaMor) [reply sdl-compliance-pipeline.yml:37]: Please see this doc: [Recommended hardware resources for running CodeQL](https://docs.github.com/en/enterprise-cloud@latest/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/recommended-hardware-resources-for-running-codeql)

**Review by iankingori — DISMISSED** [review]

**Review by CharityKathure — APPROVED** [review]

File: sdl-compliance-pipeline.yml (hunk 1 above)
Comment (TinaMor) [reply sdl-compliance-pipeline.yml]: Updated to use 8GB

**Review by iankingori — APPROVED** [review]

**Review by CharityKathure — APPROVED** [review]
LGTM

**Pull request merged**